scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+
+In every release cut, the key specified in `scanner/VULNERABILITY_VERSION` will be updated in the mapping to the current release reference (e.g., `release/4.6.0`).  Notice that the key is still updated if the version is the same.   If it's new, it will be added, and the previous version will stay.
+
+For example, in release cut 4.6 we would look into `scanner/VULNERABILITY_VERSION`, let's say its contents are:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v1
+```
+
+The mapping should be updated so `v1` points to the current release branch reference.  Let's say the mapping was:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.5.0`
+
+Then, `v1` is updated to the new release (in this example, `4.6.0`):
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.6.0`
+
+If in 4.6 the bundle version was updated to:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v2
+```
+
+On 4.6.0 release cut, the final mapping would be:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.5.0`
+`v2` | `release/4.6.0`
+
+#### Offline mode
+
+Per-release offline bundles will continue to be published.  They will be versioned by StackRox releases, but they will not be based on Y-stream (`4.5`) but on Z-stream patch releases (`4.5.0`).  The bundle will contain the correct vulnerability bundle for that particular Z-stream release.  The `manifest.json` file will continue to have a "version" attribute pointing out the Y-stream for that offline bundle, but two other fields will be added: "vulnerability_version" and "release_versions".  For example, for patch release `4.6.1` the offline bundle could be:
+
+```
+{
+  "version": "4.6",
+  "created": "2024-08-20T21:00:30+00:00",
+  "release_versions": ["4.6.0", "4.6.1", ...],
+  "vulnerability_version": "v1"
+}
+```
+
+The central definitions handler will validate the offline bundle POST against `release_versions` to ensure the running Central version is supported.  And, when consuming the offline bundle, against "vulnerability_version", guaranteeing the bundle supports the Scanner requested schema version.
+
+We will continue to offer the latest offline bundles.  They will contain all offline bundles versioned by schema, and the existing `4.5` bundle to continue to be added for backward compatibility.  Example:
+
+```
+unzip -l scanner-vuln-updates.zip                                                                             [~]
+Archive:  scanner-vuln-updates.zip
+  Length      Date    Time    Name
+---------  ---------- -----   ----
+171516200  2024-08-20 21:00   scanner-defs.zip
+   248088  2024-08-20 21:00   k8s-istio.zip
+191673607  2024-08-20 21:00   scanner-v4-defs-v1.zip
+191673030  2024-08-20 21:00   scanner-v4-defs-v2.zip
+191673030  2024-08-20 21:00   scanner-v4-defs-4.5.zip
+---------                     -------
+363437895                     3 files
```

Comment:
what are these numbers in human readable format? A few MB? GB? Perhaps this is just dummy data, but I am curious what the size of this looks like

Reply:
191673030 -> 191 673 030 -> 191 million bytes -> <= 200MiB